python/pyarrow/parquet.py

```diff
@@ -1407,3 +1406,52 @@ def read_schema(where, memory_map=False):
     schema : pyarrow.Schema
     """
     return ParquetFile(where, memory_map=memory_map).schema.to_arrow_schema()
+
+
+def get_statistics(pf):
+    """
+    Returns a DataFrame of min/max statistics cast to logical type
+
+    Parameters
+    ----------
+    pf : Parquet Dataset
+
+    Returns
+    -------
+    df : Pandas DataFrame of min, max, and logical type, indexed by column
+
+
+    """
+    import pandas as pd
+    from numpy import datetime64
+
+    df = pd.DataFrame(columns=['column', 'cmin', 'cmax', 'logical_type'])
+    df = df.set_index('column')
+
+    piece = pf.pieces[0]
+    md = piece.get_metadata(ParquetFile)
+    ps = ParquetSchema(md)
+
+    rg = md.row_group(0)
+
+    for i in range(rg.num_columns):
+        if ps.column(i).logical_type == 'UTF8':
+            lmin = rg.column(i).statistics.min.decode('UTF8')
+            lmax = rg.column(i).statistics.max.decode('UTF8')
+            df.loc[i] = [lmin, lmax, 'UTF8']
+
+        elif ps.column(i).logical_type == 'NONE':
+            lmin = rg.column(i).statistics.min
+            lmax = rg.column(i).statistics.max
+            df.loc[i] = [lmin, lmax, 'NONE']
+
+        elif ps.column(i).logical_type == 'TIMESTAMP_MICROS':
+            lmin = datetime64(rg.column(i).statistics.min, 'us')
+            lmax = datetime64(rg.column(i).statistics.max, 'us')
+            df.loc[i] = [lmin, lmax, 'TIMESTAMP_MICROS']
+        else:
+            lmin = rg.column(i).statistics.min
+            lmax = rg.column(i).statistics.max
+            df.loc[i] = [lmin, lmax, 'UNKNOWN']
```

Comment:
This seems straightforward, but I have two concerns:

-  Are there other types that we would also need to convert?
-  Is there already some place in the Arrow codebase that does this kind of logic for us,  but in a centralized way that would be guaranteed to track changes in the future?

I'm concerned that this approach would require constant changes in the future, which would increase friction between Arrow and downstream libraries, and would increase maintenance burden in the future.

This was discussed here: https://issues.apache.org/jira/browse/ARROW-4139?focusedCommentId=16823025&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16823025